DependencyInjection/M6WebKafkaExtension.php

```diff
@@ -0,0 +1,206 @@
+<?php
+declare(strict_types = 1);
+
+namespace M6Web\Bundle\KafkaBundle\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\DependencyInjection\Extension\Extension;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
+use Symfony\Component\DependencyInjection\Definition;
+
+/**
+ * Class M6WebKafkaExtension
+ * @package M6Web\Bundle\KafkaBundle\DependencyInjection
+ *
+ * This is the class that loads and manages your bundle configuration
+ */
+class M6WebKafkaExtension extends Extension
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function load(array $configs, ContainerBuilder $container)
+    {
+        $configuration = new Configuration();
+        $config = $this->processConfiguration($configuration, $configs);
+
+        $loader = new YamlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
+        $loader->load('services.yml');
+
+        $this->loadProducers($container, $config);
+        $this->loadConsumers($container, $config);
+    }
+
+    /**
+     * @param string $className
+     * @return Definition
+     */
+    public function getDefinition(string $className): Definition
+    {
+        return new Definition($className);
+    }
+
+    /**
+     * @return \RdKafka\Conf
+     */
+    public function getRdKafkaConf(): \RdKafka\Conf
+    {
+        return new \RdKafka\Conf();
+    }
+
+    /**
+     * @return \RdKafka\TopicConf
+     */
+    public function getTopicConf(): \RdKafka\TopicConf
+    {
+        return new \RdKafka\TopicConf();
+    }
+
+    /**
+     * @param ContainerBuilder $container
+     * @param array            $config
+     */
+    protected function loadProducers(ContainerBuilder $container, array $config)
+    {
+        foreach ($config['producers'] as $key => $producer) {
+            $producerDefinition = $this->getDefinition('M6Web\Bundle\KafkaBundle\RdKafkaProducerManager');
+
+            $this->setEventDispatcher($config, $producerDefinition);
+
+            $conf = $this->getReadyRdKafkaConf($producer['conf']);
+            $rdKafkaProducer = new \RdKafka\Producer($conf);
+
+            $producerDefinition->addMethodCall('setRdKafkaProducer', [$rdKafkaProducer]);
+            $this->setProducerManager($producer, $producerDefinition);
+
+            $container->setDefinition(
+                sprintf('m6_web_kafka.producer.%s', $key),
+                $producerDefinition
+            );
+        }
+    }
+
+    /**
+     * @param ContainerBuilder $container
+     * @param array $config
+     */
+    protected function loadConsumers(ContainerBuilder $container, array $config)
+    {
+        foreach ($config['consumers'] as $key => $consumer) {
+            $consumerDefinition = $this->getDefinition('M6Web\Bundle\KafkaBundle\RdKafkaConsumerManager');
```

Comment:
@fabdsp It was possible. Then, I thought it was weird. So I removed this option. Can we discuss about this point?